src/swap/components/SwapProvider.test.tsx

```diff
@@ -326,29 +310,87 @@ describe('SwapProvider', () => {
     expect(result.current.to.amount).toBe('');
   });
 
-  it('should handle quote error and reset loading state', async () => {
-    vi.mocked(getSwapQuote).mockRejectedValueOnce(new Error('Quote error'));
+  it('should setLifeCycleStatus to error when getSwapQuote throws an error', async () => {
+    const mockError = new Error('Test error');
+    vi.mocked(getSwapQuote).mockRejectedValueOnce(mockError);
     const { result } = renderHook(() => useSwapContext(), { wrapper });
     await act(async () => {
-      await result.current.handleAmountChange(
-        'from',
-        '10',
-        ETH_TOKEN,
-        DEGEN_TOKEN,
-      );
+      result.current.handleAmountChange('from', '10', ETH_TOKEN, DEGEN_TOKEN);
+    });
+    expect(result.current.lifeCycleStatus).toEqual({
+      statusName: 'error',
+      statusData: {
+        code: 'TmSPc01',
+        error: JSON.stringify(mockError),
+        message: '',
+      },
     });
-    expect(result.current.error?.quoteError).toBeDefined();
-    expect(result.current.to.loading).toBe(false);
   });
 
-  beforeEach(async () => {
-    vi.resetAllMocks();
+  it('should setLifeCycleStatus to error when getSwapQuote returns an error', async () => {
+    vi.mocked(getSwapQuote).mockResolvedValueOnce({
+      code: getSwapErrorCode('uncaught-quote'),
+      error: 'Something went wrong',
+      message: '',
+    });
+    const { result } = renderHook(() => useSwapContext(), { wrapper });
     await act(async () => {
-      renderWithProviders(() => null);
+      result.current.handleAmountChange('from', '10', ETH_TOKEN, DEGEN_TOKEN);
+    });
+    expect(result.current.lifeCycleStatus).toEqual({
+      statusName: 'error',
+      statusData: {
+        code: 'UNCAUGHT_SWAP_QUOTE_ERROR',
+        error: 'Something went wrong',
+        message: '',
+      },
     });
   });
 
-  afterEach(() => {
-    vi.clearAllMocks();
+  it('should handle submit correctly', async () => {
+    await act(async () => {
+      renderWithProviders(TestSwapComponent);
+    });
+    await act(async () => {
+      fireEvent.click(screen.getByText('Swap'));
+    });
+    expect(buildSwapTransaction).toBeCalledTimes(1);
+  });
+
+  it('should setLifeCycleStatus to error when buildSwapTransaction throws an error', async () => {
+    const mockError = new Error('Test error');
+    vi.mocked(buildSwapTransaction).mockRejectedValueOnce(mockError);
+    renderWithProviders(TestSwapComponent);
+    fireEvent.click(screen.getByText('Swap'));
+    await waitFor(() => {
+      expect(
+        screen.getByTestId('context-value-lifeCycleStatus-statusName')
+          .textContent,
+      ).toBe('error');
+      expect(
+        screen.getByTestId('context-value-lifeCycleStatus-statusData-code')
+          .textContent,
+      ).toBe('TmSPc02');
+    });
+  });
+
+  it('should setLifeCycleStatus to error when buildSwapTransaction returns an error', async () => {
```

Comment:
Adding more tests and re-organizing the tests as the order on where the code is located.